### PR TITLE
Add kuttl tests for oneagent modes

### PR DESCRIPTION
**Pull request by 0sewa0**
# Description

Some simple kuttl tests to start us off.

1 TestSuite for the oneagent modes.

Reasons KIND might fail for you: (its not the default way for this reason)
- `KIND` needs a lot of RAM for the tests to run and the pods to not be stuck in `Pending`

## How can this be tested?
PLS read the readme first: `src/testing/kuttl/README.md`
The following commands will use the cluster that your kubeconfig is pointing to.
`make kuttl-oneagent`
or
`kubectl kuttl test --config src/testing/kuttl/oneagent/oneagent-test.yaml` (you need to deploy the operator by hand in this case)

## Checklist
- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly

### CODE REVIEW

**Comment by chrismuellner** [comment]
Please add a readme with instructions how to use these tests and kuttl in the right folder

File: hack/e2e/install-kuttl.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+sudo curl -Lo /usr/local/bin/kubectl-kuttl https://github.com/kudobuilder/kuttl/releases/download/v0.11.1/kubectl-kuttl_0.11.1_linux_x86_64
+sudo chmod +x /usr/local/bin/kubectl-kuttl
+export PATH=$PATH:/usr/local/bin
```
Comment (meik99) [inline hack/e2e/install-kuttl.sh:5]:
```suggestion
#!/bin/bash

if ! kubectl kuttl; then
  sudo curl -Lo /usr/local/bin/kubectl-kuttl https://github.com/kudobuilder/kuttl/releases/download/v0.11.1/kubectl-kuttl_0.11.1_linux_x86_64
  sudo chmod +x /usr/local/bin/kubectl-kuttl
fi
```

File: Makefile [hunk 2]
```diff
@@ -59,6 +59,14 @@ helm-test:
 helm-lint:
 	cd config/helm && ./testing/lint.sh
 
+kuttl-oneagent: deploy
```
Comment (meik99) [inline Makefile:62]:
```suggestion
kuttl: 
    hack/e2e/install-kuttl.sh

kuttl-oneagent: kuttl deploy
```

File: hack/e2e/install-kuttl.sh (hunk 1 above)
Comment (meik99) [reply hack/e2e/install-kuttl.sh:5]: also this file needs to be `chmod +x`-ed to add executable permissions

**Review by meik99 — CHANGES_REQUESTED** [review]

File: hack/e2e/install-kuttl.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+if ! kubectl kuttl; then
+  sudo curl -Lo /usr/local/bin/kubectl-kuttl https://github.com/kudobuilder/kuttl/releases/download/v0.11.1/kubectl-kuttl_0.11.1_linux_x86_64
+  sudo chmod +x /usr/local/bin/kubectl-kuttl
+fi
```
Comment (gkrenn) [inline hack/e2e/install-kuttl.sh:5]: Pls add support form MacOS: 

```#!/bin/bash

if [[ "$OSTYPE" == "darwin"* ]]; then
  arch=`uname -m`
  ostype=darwin
else
  arch=linux
  ostype=linux
fi

version=0.11.1
installurl=https://github.com/kudobuilder/kuttl/releases/download/v${version}/kubectl-kuttl_${version}_${ostype}_${arch}

if ! kubectl kuttl; then
  echo Download from: $installurl
  sudo curl -Lo /usr/local/bin/kubectl-kuttl $installurl
  sudo chmod +x /usr/local/bin/kubectl-kuttl
fi
```

File: src/testing/kuttl/create-dynakube-base.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  missingVars=1
+fi
+if [[ -z "${PAASTOKEN}" ]]; then
+  echo "environment variable PAASTOKEN not set"
+  missingVars=1
+fi
+if [[ $missingVars = 1 ]]; then
+  exit 1
+fi
+
```
Comment (gkrenn) [inline src/testing/kuttl/create-dynakube-base.sh:18]: Would add check for ns: `kubectl create ns dynakube`

Comment (gkrenn) [reply src/testing/kuttl/create-dynakube-base.sh:18]: Ah, sorry, misunderstood, ignore this comment

**Review by meik99 — APPROVED** [review]

**Pull request merged**